cmd/vclusterctl/cmd/platform/list/utils.go

```diff
@@ -0,0 +1,66 @@
+package list
+
+import (
+	"encoding/json"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/log/table"
+	"github.com/sirupsen/logrus"
+)
+
+func printJSON(logger log.Logger, value []map[string]string) error {
+	bytes, err := json.MarshalIndent(value, "", "    ")
+	if err != nil {
+		return err
+	}
+	logger.WriteString(logrus.InfoLevel, string(bytes)+"\n")
+	return nil
+}
+
+// PrintData is a generic function that prints data in different formats (JSON or table).
+// It takes a logger for output, an output type (json/table), a list of headers, a slice of items,
+// and a function to extract values from each item.
+func PrintData[T any](logger log.Logger, outputType string, headers []string, items []T, getValues func(T) []string) error {
```

Comment:
Can we add a unit test for this function?
It should be easy to pass a file logger (logging to temp file) to `PrintData` and then assert on its content

Reply:
Done